include/aws/mqtt/client.h

```diff
@@ -110,6 +110,40 @@ struct aws_mqtt_topic_subscription {
     void *on_publish_ud;
 };
 
+/**
+ * host_name                 The server name to connect to. This resource may be freed immediately on return.
+ * port                      The port on the server to connect to
+ * client_id                 The clientid to place in the CONNECT packet.
+ * socket_options            The socket options to pass to the aws_client_bootstrap functions.
+ *                           This is copied into the connection
+ * tls_options               TLS settings to use when opening a connection.
+ *                           This is copied into the connection
+ *                           Pass NULL to connect without TLS (NOT RECOMMENDED)
+ * clean_session             True to discard all server session data and start fresh
+ * keep_alive_time           The keep alive value to place in the CONNECT PACKET, a PING will automatically
+ *                           be sent at this interval as well. If you specify 0, defaults will be used
+ *                           and a ping will be sent every 60s. If you're tempted to set this really low
+ *                           to find out about connection drops quickly, this is going to consume a lot of cpu.
+ *                           Instead, consider setting tcp keep alive to low timeout values and leaving the ping
+ *                           at the max value allowed by the broker.
+ * request_timeout_ms        Amount of time before a PING is considered timed out. Default is 3s.
```

Comment:
If this only applies to pings maybe update name to "ping_timeout_ms"  If it applies to more than pings than maybe adjust comment.  Related, are pings the only request/response component of mqtt?

Reply:
I don't know all the ways they apply because I only touched PING, but in general, there's probably a timeout (at least in the future) on suback and puback.